src/main/webapp/scripts/forum-script.js

```diff
@@ -29,18 +33,39 @@ function getFilters() {
 
 /**
  *  Populates the replies for a forum element with the given id
- *  in the placeholder given
+ *  in the placeholder given with no search keywords
  *
  *  @param {S.fn.init} placeholder the div that will hold the forum elements
  *  @param {long} id the long that identifies the parent of the forum
  *  elements
  */
 function expandForum(placeholder, id) {
+  // Call expandForumWithSearch with the search parameter as null so that
+  // no search is considered
+  expandForumWithSearch(placeholder, id, null);
+}
+
+/**
+ *  Populates the replies for a forum element with the given id
+ *  in the placeholder given also based on search keywords if present
+ *
+ *  @param {S.fn.init} placeholder the div that will hold the forum elements
+ *  @param {long} id the long that identifies the parent of the forum
+ *  elements
+ *  @param {String} search search text from user if applicable
+ */
+function expandForumWithSearch(placeholder, id, search) {
   placeholder.empty();
   fetch('/forum?id=' + id.toString())
       .then((response) => (response.json())).then((elements) => {
         for (let i = 0; i < elements.length; i++) {
-          createForumElement(placeholder, elements[i]);
+          if (search) {
+            if (containsSearch(elements[i].text, search)) {
+              createForumElement(placeholder, elements[i]);
+            }
+          } else {
+            createForumElement(placeholder, elements[i]);
+          }
```

Comment:
You could simplify this whole block (lines 62-68) with:
```
if (!search || containsSearch(elements[i].text, search)) {
  createForumElement(placeholder, elements[i]);
}
```

Reply:
Updated in "simplify if statement mike comment" commit